packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx

```diff
@@ -6,44 +6,53 @@ import { Checkbox } from '../../Checkbox'
 import { FacetFilterHeader } from '../FacetFilterHeader'
 import { EnumFacetFilterOption } from './EnumFacetFilterOption'
 import EnumFacetFilterDropdown from './EnumFacetFilterDropdown'
+import { SetOptional } from 'type-fest'
 
 const MAX_ENUMERATION_VALUES_TO_SHOW = 5
 
-export type RenderedFacetValue = FacetColumnResultValueCount & {
+export type RenderedFacetValue<TValue = string> = Omit<
+  SetOptional<FacetColumnResultValueCount, 'count'>,
+  'value'
+> & {
+  value: TValue
   /* Text displayed for the option, and also used to filter in search */
   displayText: string
 }
 
-export type EnumFacetFilterUIProps = {
+export type EnumFacetFilterUIProps<TValue = string> = {
   /* The title of the faceted column to be displayed */
   facetTitle: string
   /* List of all facet values and information associated with each value */
-  facetValues: RenderedFacetValue[]
-  /* Whether the 'All' checkbox should be selected. */
-  allIsSelected: boolean // allIsSelected cannot be derived from facetValues[*].isSelected. See PORTALS-2680
+  facetValues: RenderedFacetValue<TValue>[]
+  /* Determines how the component is styled and whether the 'All' checkbox should be selected. */
+  filterIsActive: boolean // filterIsActive cannot be derived from facetValues[*].isSelected. See PORTALS-2680
```

Comment:
`allIsSelected` kept confusing me because it doesn't always mean all values are selected...sometimes it means no values are selected. Hopefully you agree `filterIsActive` is less confusing--note that this is inverted from `allIsSelected`, so all places it was set and used had to be changed

Reply:
Agreed!